### PR TITLE
No longer allow to trash document templates.

### PR DESCRIPTION
**Pull request by tinagerber**
Changes in this PR:
- Change templatefolder workflow so that content can no longer be trashed
- Untrash all documents in the template folder that are trashed
Jira: https://4teamwork.atlassian.net/browse/NE-411

## Checklist (Must have)

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

## Checklist (if applicable)

_Only applicable should be left and checked._

- Upgrade steps (changes in profile):
  - [ ] Make it deferrable if possible
  - [x] Execute as much as possible conditionally

### CODE REVIEW

File: opengever/core/upgrades/20201223134632_do_not_allow_to_trash_content_in_templatefolder/upgrade.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+                                      reindex_security=False)
+
+    def delete_trashed_documents_in_template_folder(self):
+        catalog = api.portal.get_tool('portal_catalog')
+        portal_path = '/'.join(api.portal.get().getPhysicalPath())
+        templatefolder_query = {'path': {'query': portal_path, 'depth': 1},
+                                'portal_type': 'opengever.dossier.templatefolder'}
+        templatefolder_paths = [brain.getPath() for brain in catalog(templatefolder_query)]
+        for templatefolder_path in templatefolder_paths:
+            document_query = {'path': {'query': portal_path},
```
Comment (njohner) [inline opengever/core/upgrades/20201223134632_do_not_allow_to_trash_content_in_templatefolder/upgrade.py:22]: euh you are deleting all trashed documents, not just the ones in the templatefolders... I guess you wanted to use `templatefolder_path`

File: opengever/core/upgrades/20201223134632_do_not_allow_to_trash_content_in_templatefolder/upgrade.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                              'object_provides': 'opengever.document.behaviors.IBaseDocument',
+                              'trashed': True}
+            for obj in self.objects(document_query, 'Delete trashed documents in templatefolder'):
+                obj.aq_parent.manage_delObjects([obj.getId()])
+                print('Deleted obj')
```
Comment (njohner) [inline opengever/core/upgrades/20201223134632_do_not_allow_to_trash_content_in_templatefolder/upgrade.py:27]: Looks like a leftover

File: opengever/core/upgrades/20201223134632_do_not_allow_to_trash_content_in_templatefolder/upgrade.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        portal_path = '/'.join(api.portal.get().getPhysicalPath())
+        templatefolder_query = {'path': {'query': portal_path, 'depth': 1},
+                                'portal_type': 'opengever.dossier.templatefolder'}
+        templatefolder_paths = [brain.getPath() for brain in catalog(templatefolder_query)]
+        for templatefolder_path in templatefolder_paths:
+            document_query = {'path': {'query': portal_path},
+                              'object_provides': 'opengever.document.behaviors.IBaseDocument',
+                              'trashed': True}
+            for obj in self.objects(document_query, 'Delete trashed documents in templatefolder'):
+                obj.aq_parent.manage_delObjects([obj.getId()])
```
Comment (njohner) [inline opengever/core/upgrades/20201223134632_do_not_allow_to_trash_content_in_templatefolder/upgrade.py:26]: you could use `api.content.delete`...

**Review by njohner — COMMENTED** [review]
I'm really not sure we want an upgrade step deleting documents.
Can you have a look at how many such trashed templates there are, or rather how many customers actually have trashed document templates? If the number is small enough, I'd rather opt for a maintenance script for the deletion.

**Comment by njohner** [comment]
I also quickly discussed with @deiferni . On suggestion is to untrash the trashed template documents instead of deleting them, and then inform the client to check whether he wants to delete them or not.
But basically we need some sort of assessment of the current situation so that we can decide how to proceed...

**Comment by tinagerber** [comment]
@njohner I checked the SaaS deployments on the ipet server, there are 4 deployments that have trashed documents in the template folder, so it probably affects a couple of deployments.

**Comment by njohner** [comment]
I've checked Koeniz (0), Muri (1) and HBA (0), HEPVS(1), AI (140), Nidau (35), fmh (0)

**Comment by njohner** [comment]
@tinagerber We had a quick chat with @deiferni and we don't want to delete the documents in an upgrade step. Instead untrash them. And please add a note in the version specific points that we should check before the update whether there are such documents and if there are a lot, discuss with the client in advance to clean that up (maybe delete them before the update).

**Comment by tinagerber** [comment]
@njohner I untrashed the documents now in the upgrade step and added a note in https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/123077675/GEVER+Releases+Versions-Spezifische+Punkte

**Review by njohner — APPROVED** [review]
LGTM

**Pull request merged**